File: src/main/java/com/partymakers/shareparty/domain/party/usecase/impl/GetPartyExpensesImpl.java
package com.partymakers.shareparty.domain.party.usecase.impl;

import com.partymakers.shareparty.domain.party.entity.Expense;
import com.partymakers.shareparty.domain.party.entity.PartyRoom;
import com.partymakers.shareparty.domain.party.port.PartyRoomRepository;
import com.partymakers.shareparty.domain.party.usecase.GetPartyExpenses;
import com.partymakers.shareparty.domain.party.usecase.exception.NotFoundException;

import java.util.List;
import java.util.Optional;

import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public class GetPartyExpensesImpl implements GetPartyExpenses {

    private final PartyRoomRepository repository;

    @Override
    public List<Expense> getPartyExpenses(Long partyId) {
        return repository.findById(partyId).map(partyRoom -> partyRoom.getExpenses())
            .orElseThrow(NotFoundException::new);
    }
}

File: src/main/kotlin/com/partymakers/shareparty/domain/party/usecase/impl/GetPartyExpensesImpl.kt
package com.partymakers.shareparty.domain.party.usecase.impl

import com.partymakers.shareparty.domain.party.entity.Expense
import com.partymakers.shareparty.domain.party.port.PartyRoomRepository
import com.partymakers.shareparty.domain.party.usecase.GetPartyExpenses
import com.partymakers.shareparty.domain.party.usecase.exception.NotFoundException
import org.springframework.stereotype.Service

@Service
class GetPartyExpensesImpl(
    private val repository: PartyRoomRepository
) : GetPartyExpenses {

    override fun getPartyExpenses(partyId: Long): List<Expense> =
        repository.findById(partyId)
            .map { it.expenses }
            .orElseThrow { NotFoundException("Party room with id: $partyId not found") }
} 